lib/core/framework/backend.ts

```diff
@@ -0,0 +1,490 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2020 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as _ from 'lodash';
+
+import * as Kuzzle from '../../kuzzle';
+import * as Plugin from '../plugin/plugin';
+import * as kerror from '../../kerror';
+import { kebabCase } from '../../util/inflector';
+
+import { ObjectWithStringKey } from '../../util/interfaces';
+
+const assertionError = kerror.wrap('plugin', 'assert');
+const runtimeError = kerror.wrap('plugin', 'runtime');
+
+/* PipeManager class ======================================================== */
```

Comment:
(open to debate) I'm not fond of that kind of comment. In my mind, this raises an alarm saying "this needs to be split in multiple files"

Reply:
I didn't wanted to create new classes at the beginning because those classes are only kind of set to edit internal properties of the `Backend` class but I decided to create classes so we can benefits from the strong typing.
I can split them in file but we will have 6 additional files containing a class with only one or 2 methods